hypertrace-gradle-docker-java-application-plugin/src/main/java/org/hypertrace/gradle/docker/HypertraceDockerJavaApplicationPlugin.java

```diff
@@ -180,7 +188,13 @@ private void createDockerStartScriptTask(Project project) {
              startScript.setApplicationName(javaApplication.getApplicationName());
              startScript.setOutputDir(contextDir.file(DOCKER_BUILD_CONTEXT_SCRIPTS_DIR)
                                                 .getAsFile());
-             startScript.setDefaultJvmOpts(javaApplication.getApplicationDefaultJvmArgs());
+             List<String> jvmArgs = new ArrayList<>();
+             javaApplication.getApplicationDefaultJvmArgs().forEach(jvmArgs::add);
+             // add generational zgc if it is java 21 based
+             if (getHypertraceDockerApplicationExtension(project).javaVersion.get() == JavaVersion.VERSION_21) {
```

Comment:
isn't it lazy as it is already in configuration action which takes in start script as parameter?

Reply:
unfolding it a bit - yes and no.

So strictly speaking, you're right - because this uses the register api, we've deferred this action from eval time to config time. But by explicitly calling .get() we're still dependent on the order of configuration (which should be fine for our purposes). Ideally though, we want to leave it up to gradle to do the resolution when it needs the value - the last possible moment.

I wouldn't spend hours trying to fix this, but it's worth a few minutes. Basically - does setDefaultJvmOpts accept a provider of args, or does it have to be an iterable? If it takes a provider, we can leave this lazy and map the java version to the list

Reply:
got it, but seems like it only takes in iterable instead of provider

<img width="1156" alt="image" src="https://github.com/user-attachments/assets/ee90735e-b85c-4894-b43f-25e0e394b60e">